lib/galaxy/jobs/runners/util/cli/shell/local.py

```diff
@@ -37,12 +40,16 @@ class LocalShell(BaseShellExec):
     True
     >>> exec_result.returncode == TIMEOUT_RETURN_CODE
     True
+    >>> shell.execute('echo hi').stdout == "hi\\n"
+    True
     """
 
     def __init__(self, **kwds):
         pass
 
     def execute(self, cmd, persist=False, timeout=DEFAULT_TIMEOUT, timeout_check_interval=DEFAULT_TIMEOUT_CHECK_INTERVAL, **kwds):
+        if isinstance(cmd, (six.string_types, six.text_type)):
+            cmd = shlex.split(cmd)
         outf = TemporaryFile()
         p = Popen(cmd, stdin=None, stdout=outf, stderr=PIPE)
```

Comment:
Maybe replace these 4 lines with:
```python
        is_cmd_string = isinstance(cmd, six.string_types)
        outf = TemporaryFile()
        p = Popen(cmd, stdin=None, stdout=outf, stderr=PIPE, shell=is_cmd_string)
```

Reply:
Thanks, that is much better, I've rebased this with your suggestion.